### PR TITLE
WIP: ON HOLD: Allchannel 2.0

### PR DESCRIPTION
**Pull request by qstokkink**
Scalable magnet link based channels prototype.

TODO:

 - [x] Unit tests
 - [x] Load in `LaunchManyCore`
 - [ ] Switch all channel logic from AllChannel 1.0 -> 2.0
 - [ ] GUI control (*do we even need to change anything in the GUI?*)
 - [ ] Create upgrade script from AllChannel 1.0 -> 2.0
 - [ ] Remove AllChannel 1.0

### CODE REVIEW

**Comment by qstokkink** [comment]
Tests appear to be unstable, fixed in #3491.

**Comment by qstokkink** [comment]
Now that that is merged, there seems to be a process running on Windows 😢

**Comment by qstokkink** [comment]
retest this please

**Comment by qstokkink** [comment]
retest this please

**Comment by qstokkink** [comment]
As #3923 is nearing completion, I think this can be closed now.

File: Tribler/community/allchannel2/structures.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
+import os
+
+from libtorrent import add_files, bdecode, bencode, create_torrent, file_storage, set_piece_hashes
+
+
+PIECE_SIZE = 16*1024*1024 # 16 MB: Holds 762600 magnetlinks without metadata
```
Comment (synctext) [inline Tribler/community/allchannel2/structures.py:6]: Unclear to me what this does. Is this re-doing TFTP somewhat?

Comment (devos50) [reply Tribler/community/allchannel2/structures.py:6]: This most likely refers to the torrent piece size, which seems to be fixed to 16MB for this implementation.

**Pull request closed without merging**